### PR TITLE
Support DD rpm loading from local disk device (#1377233)

### PR DESCRIPTION
**Pull request by jkonecny12**
You can now use inst.dd=hd:LABEL=DD:/dd.rpm as kernel boot option.

Changes:
driver-updates-genrules.sh - parse LABEL=DD and similar correctly

driver_updates.py
Optional parameter RPMPATH on --disk param, this parameter is path to the rpm file. If this parameter is present mount device and process the rpm on the mounted device like usual.

*Resolves: rhbz#1377233*

### CODE REVIEW

**Comment by Frodox** [comment]
What is meaning of dd ..?

**Comment by jkonecny12** [comment]
As I already answered on IRC `Driver Disk` :)

File: dracut/driver_updates.py [hunk 1]
```diff
@@ -491,13 +492,25 @@ def _process_driver_disk(dev, interactive=False):
...
-        return _process_driver_rpm(rpm)
+        if dev:
+            return _mount_and_process_driver_rpm(rpm, dev)
+        else:
+            return _process_driver_rpm(rpm)
     except (subprocess.CalledProcessError, IOError) as e:
         log.error("ERROR: %s", e)
         return {}
 
+def _mount_and_process_driver_rpm(rpm, dev):
```
Comment (M4rtinK) [inline dracut/driver_updates.py]: If I read this correctly, the `mounted` context managed unmounts the device once the block is processed, right ? In such case the name of the function (`_mount_and_process_driver_rpm`) seems misleading - it sounds like the device remains mounted after the function has been called.

Sp I recommend changing changing the name to something like `_process_driver_rpm_from_device()`. :)

File: dracut/driver_updates.py [hunk 2]
```diff
@@ -491,13 +492,25 @@ def _process_driver_disk(dev, interactive=False):
 
     return modules
 
-def process_driver_rpm(rpm):
+def process_driver_rpm(rpm, dev = None):
```
Comment (rvykydal) [inline dracut/driver_updates.py]: I'd just leave out the spaces in default argument for consistency (dev=None).
Sorry I don't have any better remark to the patch, looks good to me.

File: dracut/driver_updates.py (hunk 1 above)
Comment (jkonecny12) [inline dracut/driver_updates.py]: Good point, fixed.

File: dracut/driver_updates.py (hunk 2 above)
Comment (jkonecny12) [inline dracut/driver_updates.py]: Thanks, fixed.
No need to be sorry for this, that is always a good mark.

**Pull request merged**